src/sdp_proxy/discovery_proxy.cpp

```diff
@@ -284,8 +282,13 @@ void DiscoveryProxy::OnHostDiscovered(const std::string
         {
             continue;
         }
+
         splitError = SplitFullHostName(queryName, hostName, domain);
-        assert(splitError == OTBR_ERROR_NONE);
+
+        if (splitError != OTBR_ERROR_NONE)
+        {
+            continue;
```

Comment:
nit: it may worth a debug or info log for triage

Reply:
Can be a good idea but may also extra logs (if there are many queries with invalid format).

I leave it to you or other folks to add logs in future PRs.

Reply:
The issue is that if the client is doing a resolve, but we get a service published at the time, the path of:

```
       case OT_DNSSD_QUERY_TYPE_BROWSE:
            splitError = SplitFullServiceName(queryName, serviceName, domain);
```

will get hit, but that will (and should) always fail since the request was for an instance, not a service.